src/widgets/keyboard/lv_keyboard.c

```diff
@@ -388,14 +899,147 @@ static void lv_keyboard_constructor(const lv_obj_class_t * class_p, lv_obj_t * o
     keyboard->ta         = NULL;
     keyboard->mode       = LV_KEYBOARD_MODE_TEXT_LOWER;
     keyboard->popovers   = 0;
+#if LV_USE_KEYBOARD_PINYIN
+    keyboard->dict = NULL;
+    keyboard->candidate_num = 0;
+    keyboard->candidate_btnm_map = NULL;
+#if LV_KEYBOARD_PINYIN_USE_EXT_BUTTONMATRIX
+    keyboard->ext_candidate_num = 0;
+    keyboard->ext_candidate_btnm_map = NULL;
+#endif
+#endif
 
     lv_obj_align(obj, LV_ALIGN_BOTTOM_MID, 0, 0);
     lv_obj_add_event_cb(obj, lv_keyboard_def_event_cb, LV_EVENT_VALUE_CHANGED, NULL);
     lv_obj_set_style_base_dir(obj, LV_BASE_DIR_LTR, 0);
 
+#if LV_USE_KEYBOARD_PINYIN
+#if LV_KEYBOARD_PINYIN_USE_DEFAULT_DICT
+    keyboard->dict = lv_def_pinyin_dict;
+#endif
+
+    lv_opa_t keyboard_bg_opa = lv_obj_get_style_bg_opa(obj, 0);
+    lv_color_t keyboard_bg_color = lv_obj_get_style_bg_color(obj, 0);
+
+    static lv_style_t pinyin_default_style;
+    lv_style_init(&pinyin_default_style);
```

Comment:
style should only be initialized once if I recall correctly.

Reply:
> style should only be initialized once if I recall correctly.

Is this okay?
```c
lv_color_t keyboard_bg_color = lv_obj_get_style_bg_color(obj, 0);

static lv_style_t pinyin_default_style;
if(pinyin_default_style.prop_cnt < 1) {
    lv_style_init(&pinyin_default_style);
    lv_style_set_bg_opa(&pinyin_default_style, keyboard_bg_opa);
    lv_style_set_bg_color(&pinyin_default_style, keyboard_bg_color);
    lv_style_set_border_width(&pinyin_default_style, 0);
    lv_style_set_pad_all(&pinyin_default_style, 0);
    lv_style_set_pad_gap(&pinyin_default_style, 0);
    lv_style_set_radius(&pinyin_default_style, 0);
#if LV_FONT_SIMSUN_16_CJK
    lv_style_set_text_font(&pinyin_default_style, &lv_font_simsun_16_cjk);
#endif
}
```

Reply:
Can we make this style the [default theme](https://github.com/lvgl/lvgl/blob/master/src/themes/default/lv_theme_default.c#L1041)?

Reply:
> Can we move this style to the [default theme](https://github.com/lvgl/lvgl/blob/master/src/themes/default/lv_theme_default.c#L1041)?

Can anyone please confirm this problem? If not, I will modify it according to the previous plan.